extconf.rb

```diff
@@ -120,8 +120,15 @@ def rule(target, clean = nil)
 end
 
 case version_str = `#{pg_config} --version`
-when /^PostgreSQL ([7-9])\.([0-9]{1,3})(\.[0-9]{1,3})?$/
-   version = 10 * $1.to_i + $2.to_i
+when /^PostgreSQL ([7-9]|[1-9][0-9])\.([0-9]{1,3})(\.[0-9]{1,3})?$/
+   if $1.to_i < 10
+       version = 10 * $1.to_i + $2.to_i
+   else
+       # In v10+, the second number means "minor" version - which is
+       # expected to be >= 10 one day.  So rather mutiply by 100 to avoid
+       # overflow.
+       version = 100 * $1.to_i + $2.to_i
```

Comment:
This does not look right. Isn't the version just use for checking for for the major version in the code below? I think something like the below would be better:

```ruby
when /^PostgreSQL ([7-9]|[1-9][0-9])\.([0-9]{1,3})(\.[0-9]{1,3})?$/
   if $1.to_i < 10
       # Before v10 the second number was also part of the major version 
       version = 10 * $1.to_i + $2.to_i
   else
       version = 10 * $1.to_i
```

Reply:
I probably prefer not to expand `10.1 => 100`, that would look kind of weird.  We usually check like `VERSION >= NN` anways.  But I can admit it is matter of taste.  Any other opinion?  (I'd probably switch to what @jeltz suggests).

> Isn't the version just use for checking for for the major version in the code below?

Plus it is a `#define`, used in code elsewhere.

Reply:
I think we should be consistent. In the current master 9.6.1 becomes 96, and in that case 10.1 should become 100. I would be perfectly fine with including the minor version, but in the case we need to do something like 9.6.1 => 90601 and 10.1 => 100001 since we need to support 9.6.11 => 90611.

Reply:
Also why do we need that #define? PostgreSQL already defines `PG_VERSION_NUM`. I think I could submit a patch which starts to use `PG_VERSION_NUM` that after you have merged this.

Reply:
Patch for `PG_VERSION_NUM` or `PG_MAJORVERSION` would be awesome.

Reply:
Turns out such a patch would not be necessary since none of the checks check for anything newer than 8.4.

WIP commit which removes the legacy ifdefs: https://github.com/jeltz/postgresql-plruby/commit/5c65ca16fa636b502cae9b1e3404e508ce22aa6a